lib/puppet_x/puppetlabs/dsc_type_helpers.rb

```diff
@@ -126,6 +126,24 @@ def self.validate_mof_type(mof_type, embeddedinstance_name, name, value)
           end
         end
       end
+
+      def self.reboot_relationship(resource)
+        reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+        return [] if !reboot_resource
+
+        # if an edge already exists from Reboot[dsc_reboot] to this resource, we have problems
```

Comment:
@ferventcoder because it would create a circular dependency (and also doesn't make any sense)

Before we merge this, let's run a test where a user has created a bad manifest like:

```puppet
reboot { 'dsc_reboot':
  when => pending
} ->

dsc_file {'foo':
  dsc_destinationpath => 'c:\\foo.txt',
  dsc_contents => 'foo'
}
```